src/Files.App/Actions/FileSystem/FlattenRecursiveAction.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using System.IO;
+using Windows.Storage;
+using Microsoft.Extensions.Logging;
+
+namespace Files.App.Actions
+{
+	internal sealed class FlattenRecursiveAction : ObservableObject, IAction
+	{
+		private readonly IContentPageContext context;
+
+		public string Label
+			=> "FlattenRecursive".GetLocalizedResource();
+
+		public string Description
+			=> "FlattenDescription".GetLocalizedResource();
+
+		public bool IsExecutable =>
+			context.ShellPage is not null &&
+			context.HasSelection &&
+			context.SelectedItem?.PrimaryItemAttribute is StorageItemTypes.Folder;
+
+		public FlattenRecursiveAction()
+		{
+			context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+			context.PropertyChanged += Context_PropertyChanged;
+		}
+
+		public Task ExecuteAsync(object? parameter = null)
+		{
+			if (context.ShellPage?.ShellViewModel is null)
+				return Task.CompletedTask;
+
+			var items = context.SelectedItems;
+
+			if (items is null || !items.Any() || items.Any(item => !item.IsFolder))
+				return Task.CompletedTask;
+
+			foreach (var item in items)
+			{
+				FlattenFolderAsync(item.ItemPath);
+			}
+
+			return Task.CompletedTask;
+		}
```

Comment:
```suggestion
		public async Task ExecuteAsync(object? parameter = null)
		{
			if (context.ShellPage?.ShellViewModel is null)
				return;

			var items = context.SelectedItems;

			if (items is null || !items.Any() || items.Any(item => !item.IsFolder))
				return;

			await Task.WhenAll(items.Select(item => FlattenFolderAsync(item.ItemPath)));
		}
```
@0x5bfa would this be better for performance?